cylc/flow/platforms.py

```diff
@@ -387,14 +389,28 @@ def get_host_from_platform(platform, method='random'):
             - Random Selection with check for host availability
 
     """
-    if method == 'random':
-        return random.choice(platform['hosts'])
-    elif method == 'first':
-        return platform['hosts'][0]
+    from cylc.flow import LOG
+    if badhosts is not None:
```

Comment:
This shouldn't be `LOG.critical` when merged, but it makes it easier to see what is happening whilst debugging.